PR 1.6.1 release notes (#7400)
istio/istio.io: content/en/news/releases/1.6.x/announcing-1.6.1/index.md

---
title: Announcing Istio 1.6.1
linktitle: 1.6.1
subtitle: Patch Release
description: Istio 1.6.1 patch release.
publishdate: 2020-06-03
release: 1.6.1
aliases:
    - /news/announcing-1.6.1
---

This release contains bug fixes to improve robustness. This release note describes
what’s different between Istio 1.6.0 and Istio 1.6.1.

{{< relnote >}}

> 

Reviewer: ```suggestion
# Bookinfo Sample Application Security Fixes
We've updated the versions of Node and jQuery used in the Bookinfo sample application. Node has been upgraded from version 12.9 to 12.18. jQuery has been updated from version 2.1.4 to version 3.5.0.The highest rated vulnerability fixed: *HTTP request smuggling using malformed Transfer-Encoding header (Critical) (CVE-2019-15605)*
```

Reviewer 2: Nitpicky suggestion: We usually use sentence case for headings (https://developers.google.com/style/headings) rather than title case, so I would recommend changing the capitalization on this heading to this:

# Bookinfo sample application security fixes

Author: Added both suggestions

- **Fixed** release URL name for `istioctl upgrade`
- **Fixed** `k8s.overlays` for cluster resources
- **Fixed** `HTTP/HTTP2` conflict at Gateway ([Issue 24061](https://github.com/istio/istio/issues/24061) and [Issue 19690](https://github.com/istio/istio/issues/19690))
- **Fixed** Istio Operator to respect the `--operatorNamespace` argument ([Issue 24073](https://github.com/istio/istio/issues/24073))
- **Fixed** Istio operator hanging when uninstalling Istio ([Issue 24038](https://github.com/istio/istio/issues/24038))
- **Fixed** TCP metadata exchange for upstream clusters that specify `http2_protocol_options` ([Issue 23907](https://github.com/istio/istio/issues/23907))
- **Added** `sideEffects` field to `MutatingWebhookConfiguration` for `istio-sidecar-injector` ([Issue 23485](https://github.com/istio/istio/issues/23485))
- **Improved** installation for replicated control planes ([Issue 23871](https://github.com/istio/istio/issues/23871))
- **Fixed** `istioctl experimental precheck` to report compatible versions of Kubernetes (1.14-1.18) ([Issue 24132](https://github.com/istio/istio/issues/24132))

> - **Fixed** Istio operator namespace mismatches causing a resource leak when pruning resources ([Issue 24222](https://github.com/istio/istio/issues/24222))

Reviewer 3: ```suggestion
- **Fixed** Istio operator namespace mismatches that caused a resource leak when pruning resources ([Issue 24222](https://github.com/istio/istio/issues/24222))
```

- **Fixed** SDS Agent failing to start when Proxy uses file mounted certs for Gateways ([Issue 23646](https://github.com/istio/istio/issues/23646))

> - **Fixed** TCP over HTTP conflicts causing invalid configuration to be generated ([Issue 24084](https://github.com/istio/istio/issues/24084))

Reviewer 3: ```suggestion
- **Fixed** TCP over HTTP conflicts that caused invalid configuration to be generated ([Issue 24084](https://github.com/istio/istio/issues/24084))
```

> - **Fixed** the use of external name when remote pilot address is a hostname ([Issue 24155](https://github.com/istio/istio/issues/24155))

Reviewer 3: ```suggestion
- **Fixed** the use of external name when remote Pilot address is a hostname ([Issue 24155](https://github.com/istio/istio/issues/24155))
```

- **Fixed** Istio CNI node `DaemonSet` starting when Istio CNI and `cos_containerd` are enabled on Google Kubernetes Engine (GKE) ([Issue 23643](https://github.com/istio/istio/issues/23643))
- **Fixed** Istio CNI causing pod initialization to experience a 30-40 second delay on startup when DNS unreachable ([Issue 23770](https://github.com/istio/istio/issues/23770))
- **Improved** Google Stackdriver telemetry use of UIDs with GCE VMs
- **Improved** telemetry plugins to not crash due invalid configuration ([Issue 23865](https://github.com/istio/istio/issues/23865))

> - **Fixed** a proxy sidecar segfault on WASM filters that http call responds an empty body ([Issue 23890](https://github.com/istio/istio/issues/23890))

Reviewer 3: ```suggestion
- **Fixed** a proxy sidecar segfault when the response to HTTP calls by WASM filters are empty ([Issue 23890](https://github.com/istio/istio/issues/23890))
```

- **Fixed** the use of external name when remote Pilot address is a hostname ([Issue 24155](https://github.com/istio/istio/issues/24155))
- **Fixed** Istio CNI node `DaemonSet` starting when Istio CNI and `cos_containerd` are enabled on Google Kubernetes Engine (GKE) ([Issue 23643](https://github.com/istio/istio/issues/23643))
- **Fixed** Istio CNI causing pod initialization to experience a 30-40 second delay on startup when DNS unreachable ([Issue 23770](https://github.com/istio/istio/issues/23770))
- **Improved** Google Stackdriver telemetry use of UIDs with GCE VMs
- **Improved** telemetry plugins to not crash due invalid configuration ([Issue 23865](https://github.com/istio/istio/issues/23865))
- **Fixed** a proxy sidecar segfault when the response to HTTP calls by WASM filters are empty ([Issue 23890](https://github.com/istio/istio/issues/23890))
- **Fixed** a proxy sidecar segfault while parsing CEL expressions ([Issue 497](https://github.com/envoyproxy/envoy-wasm/issues/497))

# Bookinfo sample application security fixes

We've updated the versions of Node and jQuery used in the Bookinfo sample application. Node has been upgraded from

> version 12.9 to 12.18. jQuery has been updated from version 2.1.4 to version 3.5.0.The highest rated vulnerability fixed:

Reviewer 4: ```suggestion
version 12.9 to 12.18. jQuery has been updated from version 2.1.4 to version 3.5.0. The highest rated vulnerability fixed:
```
